pkgs/development/compilers/lunarml/default.nix

```
@@ -0,0 +1,68 @@
+{ lib
+, fetchFromGitHub
+, fetchzip
+, stdenvNoCC
+, mlton
+, lua5_3
+, luajit
+, nodejs
+, withExamples ? false
+}:
+
+let
+  pname = "lunarml";
+in
+stdenvNoCC.mkDerivation {
+  inherit pname;
+
+  version = "unstable-2022-01-24";
+
+  src = fetchFromGitHub {
+    owner = "minoki";
+    repo = "LunarML";
+    rev = "60b646ae7d55550626642a1b3f9a0af3944cc2a6";
+    sha256 = "sha256-lI947OQlb5s9tkSns9bP4uNyjsa1fsPuPiczgfVRe1I=";
+  };
+
+  nativeBuildInputs = [
+    mlton
+  ];
+
+  nativeCheckInputs = [
+    lua5_3
+    luajit
+    nodejs
+  ];
+
+  postPatch = ''
+    cd "thirdparty"
+    make smlnj-lib
+    make ml-yacc
+    cd ".."
+  '';
+
+  postBuild = ''
+    cd "thirdparty"
+    make install-smlnj-lib
+    make install-mlyacc-lib
+    cd ".."
+  '';
```

```suggestion
  postBuild = ''
    make -C thirdparty install
  '';
```

Should be enough? This mirrors https://github.com/minoki/LunarML/blob/44d097e665ee16a77f00410b501f7e19858ff4fe/.github/workflows/ci.yaml#L30